src/analysis/asymptotic_equivalent.lean

```diff
@@ -0,0 +1,227 @@
+(h.is_o.trans_is_O h.is_O_symm).symm
+
+@[trans] lemma is_equivalent.trans (huv : u ~[l] v) (hvw : v ~[l] w) : u ~[l] w :=
+begin
+  rw is_equivalent,
+  convert (huv.is_o.trans_is_O hvw.is_O).add hvw.is_o,
+  ext,
+  repeat {rw pi.sub_apply},
+  abel
+end
+
+lemma is_equivalent_zero_iff_eventually_zero : u ~[l] 0 ↔ u =ᶠ[l] 0 :=
+begin
+  rw [is_equivalent, sub_zero],
+  exact is_o_zero_right_iff
+end
+
+lemma is_equivalent_const_iff_tendsto {c : β} (h : c ≠ 0) : u ~[l] (λ _, c) ↔ tendsto u l (𝓝 c) :=
+begin
+  rw [is_equivalent, is_o_const_iff h],
+  split; intro h;
+  [ { have := h.add tendsto_const_nhds, rw zero_add at this },
+    { have := h.add tendsto_const_nhds, rw ← sub_self c} ];
+  convert this; ext; simp [sub_eq_add_neg]
+end
+
+lemma is_equivalent.tendsto_const {c : β} (hu : u ~[l] (λ _, c)) : tendsto u l (𝓝 c) :=
+begin
+  rcases (em $ c = 0) with ⟨rfl, h⟩,
+  { exact (tendsto_congr' $ is_equivalent_zero_iff_eventually_zero.mp hu).mpr tendsto_const_nhds },
+  { exact (is_equivalent_const_iff_tendsto h).mp hu }
+end
+
+end normed_group
+
+open_locale asymptotics
+
+section normed_field
+
+variables {α β : Type*} [normed_field β] {t u v w : α → β} {l : filter α}
+
+lemma is_equivalent_iff_exists_mul_eq : u ~[l] v ↔
+  ∃ (φ : α → β) (hφ : tendsto φ l (𝓝 1)), u =ᶠ[l] φ * v :=
+begin
+  rw [is_equivalent, is_o_iff_exists_eq_mul],
+  split; rintros ⟨φ, hφ, h⟩; [use (φ + 1), use (φ - 1)]; split,
+  { conv in (𝓝 _) { rw ← zero_add (1 : β) },
+    exact hφ.add (tendsto_const_nhds) },
+  { convert h.add (eventually_eq.refl l v); ext; simp [add_mul] },
+  { conv in (𝓝 _) { rw ← sub_self (1 : β) },
+    exact hφ.sub (tendsto_const_nhds) },
+  { convert h.sub (eventually_eq.refl l v); ext; simp [sub_mul] }
+end
+
+lemma is_equivalent.exists_mul_eq (huv : u ~[l] v) :
+  ∃ (φ : α → β) (hφ : tendsto φ l (𝓝 1)), u =ᶠ[l] φ * v :=
+is_equivalent_iff_exists_mul_eq.mp huv
+
+lemma is_equivalent_of_tendsto_one (hz : ∀ᶠ x in l, v x = 0 → u x = 0)
+  (huv : tendsto (u/v) l (𝓝 1)) : u ~[l] v :=
+begin
+  rw is_equivalent_iff_exists_mul_eq,
+  refine ⟨u/v, huv, hz.mono $ λ x hz', (div_mul_cancel_of_imp hz').symm⟩,
+end
+
+lemma is_equivalent_of_tendsto_one' (hz : ∀ x, v x = 0 → u x = 0) (huv : tendsto (u/v) l (𝓝 1)) :
+  u ~[l] v :=
+is_equivalent_of_tendsto_one (eventually_of_forall hz) huv
+
+lemma is_equivalent_iff_tendsto_one (hz : ∀ᶠ x in l, v x ≠ 0) :
+  u ~[l] v ↔ tendsto (u/v) l (𝓝 1) :=
+begin
+  split,
+  { intro hequiv,
+    have := hequiv.is_o.tendsto_0,
+    simp only [pi.sub_apply, sub_div] at this,
+    have key : tendsto (λ x, v x / v x) l (𝓝 1),
+    { exact (tendsto_congr' $ hz.mono $ λ x hnz, @div_self _ _ (v x) hnz).mpr tendsto_const_nhds },
+    convert this.add key,
+    { ext, simp },
+    { norm_num } },
+  { exact is_equivalent_of_tendsto_one (hz.mono $ λ x hnvz hz, (hnvz hz).elim) }
+end
+
+lemma is_equivalent.tendsto_nhds {c : β} (huv : u ~[l] v) (hv : tendsto u l (𝓝 c)) :
```

Comment:
This is true in any normed group: if `c ≠ 0`, then this is `is_equivalent.trans` modulo `is_equivalent_const_iff_tendsto`. Otherwise you have `huv.symm : is_o (v - u) u` and `hv' : is_o u 1 l`, hence `is_o v 1 l`.

Reply:
Oh indeed, thanks !